terraform/modules/infra-fargate/main.tf

```diff
@@ -29,6 +29,12 @@ resource "aws_lb_target_group" "lb_tg" {
   protocol    = "HTTP"
   vpc_id      = data.aws_vpc.vpc.id
   target_type = "ip"
+
+  health_check {
+    path     = "/"
+    timeout  = 30
+    interval = 60
```

Comment:
The reason for overriding the timeout and interval is due to the app starting up processes if they're both not set then the health check fails and the container is drained.

Reply:
Ah I see, thanks. Don't worry about this for now then. Later on we can look at setting the startup grace period later (`health_check_grace_period_seconds` on the aws_ecs_service resource) which should fix this without needing to mess with all the other timeouts.

Reply:
A 30 second timeout for the healtcheck request was chosen since Publisher's Redis connection timeout is ~20 seconds.

The `/healthcheck` endpoint reports Redis connectivity. If Redis is unavailable then the healthcheck endpoint will return a `200` status (with Redis errors reported in JSON) after waiting ~20 seconds for Redis.

If the timeout were <20 seconds, then Redis being unavailable to Publisher would prompt the ALB to cycle the Publisher instance, despite the application instance itself being healthy.

This is a symptom of a bigger problem: `/healthcheck` endpoints are going to be used differently in ECS. We should implement e.g. a `/v2/healthcheck` endpoint to applications during the migration.

In the Icinga/EC2 world we used `/healthcheck.json` endpoints to check for specific problems with an application and dependencies. We probably don't want the Publisher service to tell its ALBs about Redis. We should instead put Redis connectivity into a `/metrics` endpoint for Prometheus (or Statsd in the meantime) and alert on Redis connectivity from Graphite/Prometheus. We could also instead alert on symptoms like 500s when users try to enqueue publishing jobs onto the Redis-backed queue.

Reply:
Good point - I think there'll be a fair bit of work and iterating on the design of the healthchecking and app-app communication in general. The ALB (target group) healthchecks are unusual in that they seem to conflate liveness (whether to kill the container) with readiness (whether to send traffic to the container), so I think you're right about needing to ignore dependencies on the app healthcheck.

On alerting, the winning strategy is indeed to alert on symptoms (as a rule of thumb, with occasional common-sense exceptions) and set up dashboards to show up common causes easily (e.g by showing errors rates of incoming requests vs outgoing requests to dependencies).